packages/core/src/external/athenahealth/index.ts

```diff
@@ -397,6 +437,238 @@ class AthenaHealthApi {
     }
   }
 
+  async createProblem({
+    cxId,
+    patientId,
+    departmentId,
+    condition,
+  }: {
+    cxId: string;
+    patientId: string;
+    departmentId: string;
+    condition: Condition;
+  }): Promise<ProblemCreateResponse> {
+    const { log, debug } = out(
+      `AthenaHealth create problem - cxId ${cxId} practiceId ${this.practiceId} patientId ${patientId} departmentId ${departmentId}`
+    );
+    const chartProblemUrl = `/chart/${this.stripPatientId(patientId)}/problems`;
+    try {
+      const snomedCode = this.getSnomedConditionCode(condition);
+      if (!snomedCode) {
+        throw new MetriportError("No SNOMED code found for condition", undefined, {
+          cxId,
+          practiceId: this.practiceId,
+          patientId,
+          departmentId,
+          conditionId: condition.id,
+        });
+      }
+      const startDate = this.getConditionStartDate(condition);
+      if (!startDate) {
+        throw new MetriportError("No SNOMED code found for condition", undefined, {
+          cxId,
+          practiceId: this.practiceId,
+          patientId,
+          departmentId,
+          conditionId: condition.id,
+        });
+      }
+      const conditionStatus = this.getConditionStatus(condition);
+      const athenaStatus = conditionStatus
+        ? problemStatusesAthena.find(
+            status => status.toLowerCase() === conditionStatus.toLowerCase()
+          )
+        : undefined;
+      const data = {
+        departmentid: this.stripDepartmentId(departmentId),
+        note: "Added via Metriport App",
+        snomedcode: snomedCode,
+        startdate: this.formatDate(startDate),
+        status: athenaStatus,
+        THIRDPARTYUSERNAME: undefined,
+        PATIENTFACINGCALL: undefined,
+      };
+      const response = await this.axiosInstanceProprietary.post(
+        chartProblemUrl,
+        this.createDataParams(data)
+      );
+      if (!response.data) throw new MetriportError(`No body returned from ${chartProblemUrl}`);
+      debug(`${chartProblemUrl} resp: `, () => JSON.stringify(response.data));
+      if (responsesBucket) {
+        const filePath = createHivePartitionFilePath({
+          cxId,
+          patientId,
+          date: new Date(),
+        });
+        const key = `athenahealth/chart/problem/${filePath}/${uuidv7()}.json`;
```

Comment:
I'd create a function to build these key names... most of them share the same structure, so it would be good to reuse functions, like we do with the ones that build keys for documents on s3.

Reply:
Done.